cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/workflow/CustomActionExecutor.java

```diff
@@ -86,7 +86,7 @@ void execute() throws Exception {
       customActionContext.setState(new ProgramState(ProgramStatus.COMPLETED, null));
 
     } catch (Throwable t) {
-      customActionContext.setState(new ProgramState(ProgramStatus.FAILED, Throwables.getRootCause(t).getMessage()));
+      customActionContext.setState(new ProgramState(ProgramStatus.FAILED, t.getMessage()));
```

Comment:
is the exception here the direct one thrown by the custom action itself? Or does it get wrapped somehow?

Reply:
It should be the one directly thrown by the CustomAction from what I can see. CustomActionExecutor makes a call to customActionContext.execute(customAction::run) which seems to call the run() method of the customAction class. Thus, any thrown exception from the customAction will be propagated back to this method.